### PR TITLE
fix:  respect readOnly and writeOnly properties for required checks

### LINKED ISSUE #126: Readonly-required properties are being marked as breaking changes
**Describe the bug**
Oasdiff ignores `readOnly: true` for required fields marking them as breaking changes when they are not 

**To Reproduce**
base example
```yaml
openapi: 3.0.1
info:  
  version: "1.0"
servers:
- url: https://localhost:9080
paths:
  /api/v1.0/groups:    
    post:     
      operationId: createOneGroup
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/GroupView'
        description: Creates one project.
        required: true
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GroupView'
          description: OK        
      summary: Create One Project
  /api/v1.0/groups/{groupId}:
    get:      
      operationId: returnOneGroup
      parameters:      
      - $ref: '#/components/parameters/groupId'      
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GroupView'
          description: OK
      summary: Return One Project
components:
  parameters:
    groupId:      
      in: path
      name: groupId
      required: true
      schema:
        type: string        
  schemas:
    GroupView:
      type: object
      properties:        
        created:
          type: string
          format: date-time          
          readOnly: true
        id:
          type: string          
          readOnly: true        
        name:
          type: string                  
      required:    
      - name
```

revision example
```yaml
openapi: 3.0.1
info:  
  version: "1.0"
servers:
- url: https://localhost:9080
paths:
  /api/v1.0/groups:    
    post:     
      operationId: createOneGroup
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/GroupView'
        description: Creates one project.
        required: true
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GroupView'
          description: OK        
      summary: Create One Project
  /api/v1.0/groups/{groupId}:
    get:      
      operationId: returnOneGroup
      parameters:      
      - $ref: '#/components/parameters/groupId'      
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GroupView'
          description: OK
      summary: Return One Project
components:
  parameters:
    groupId:      
      in: path
      name: groupId
      required: true
      schema:
        type: string        
  schemas:
    GroupView:
      type: object
      properties:        
        created:
          type: string
          format: date-time          
          readOnly: true
        id:
          type: string          
          readOnly: true        
        name:
          type: string                  
      required:    
      - name
      - created
      - id

```

1. Run `oasdiff -breaking-only -base base.yaml -revision revision.yaml`
2. Receive 
```
    modified:
        ?   method: POST
            path: /api/atlas/v1.0/groups
        :   requestBody:
                content:
                    mediaTypeModified:
                        application/json:
                            schema:
                                required:
                                    stringsdiff:
                                        added:
                                            - id
                                            - created
```
**Expected behavior**
It should not be a diff for POST, only GET and not a breaking change 

**Screenshots**
redocly safely ignores read_only fields

Screenshot 2022-08-02 at 14 17 29

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 1.1.39

**Additional context**
From the OAS definition
> Relevant only for Schema "properties" definitions. Declares the property as "read only". This means that it MAY be sent as part of a response but SHOULD NOT be sent as part of the request. If the property is marked as readOnly being true and is in the required list, the required will take effect on the response only. A property MUST NOT be marked as both readOnly and writeOnly being true. Default value is false.

### PR DESCRIPTION
**Pull request by gssbzn**
Closes #126

Edit: the original approach was to filter only if not breaking changes but I though about it a bit better and I think it makes more sense to always filter readonly for a request type and write only for a response type since this is more aligned with the spec

### CODE REVIEW

File: data/openapi31-test2.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ components:
       required:
         - id
         - name
-        - nickname
```
Comment (gssbzn) [inline data/openapi31-test2.yaml:27]: there's no nickname property, if this was intentional need to address this at the test level

File: diff/diff_test.go [hunk 2]
```diff
@@ -16,13 +16,15 @@ const (
 )
 
 func l(t *testing.T, v int) *openapi3.T {
+	t.Helper()
```
Comment (reuvenharrison) [inline diff/diff_test.go:19]: :-)

**Comment by reuvenharrison** [comment]
Hi Gustavo,
Your contribution is great!

In addition to the scenario you addressed (adding/deleting properties from/to the RequiredProperties list), I implemented solutions for a couple more use-cases:

1. Adding/deleting properties
2. Modifying read/write only flags for existing properties

It would be great if you could take a look and verify these use-cases too.

Thanks,
Reuven

**Comment by gssbzn** [comment]
@reuvenharrison thanks for looking into this, I see you pushed some changes to address the comments and I just merged main to resolve some conflicts, is there anything else you think I should look into before getting it merge?

**Pull request merged**